eip-0041.md

```diff
@@ -0,0 +1,148 @@
+# EIP-41 Stealth Address Standard
+
+* Author: ross-weir
+* Status: Proposed
+* Created: 16-Dec-2022
+* Last edited: 17-Dec-2022
+* License: CC0
+* Forking: not needed
+
+## Motivation
+
+This specification defines a standardized way of implementing and interacting with stealth addresses on the Ergo blockchain.
+
+Stealth addresses enable recipients of a payment to remain anonymous when receiving funds thus providing financial privacy should an actor desire it.
+
+## Scenario
+
+An actor, `Receiver`, wishes to receive a stealth payment so they generate a public address and share it.
+
+An actor, `Sender`, wishes to make a stealth payment to `Receiver` so they create a box protected by a "one-time-secret" generated from the `Receiver`s public address. Due to the method of generation this box will be spendable by `Receiver` and cannot be linked to the `Receiver`s public address they shared.
+
+## Stealth address specification
+
+The implemention suggested in this EIP was posted by `scalahub` in a thread on `ergoforum.org` [[1]](#1) and is outlined below.
+
+**Script protecting stealth boxes:**
+
+```scala
+{
+    // ===== Contract Information ===== //
+    // Name: EIP-0041 Stealth address contract
+    // Version: 1.0.0
+    val gr = SELF.R4[GroupElement].get
+    val gy = SELF.R5[GroupElement].get
+    val ur = SELF.R6[GroupElement].get
+    val uy = SELF.R7[GroupElement].get
+
+    proveDHTuple(gr,gy,ur,uy)
+}
+```
+
+**Script ErgoTree:**
+
+```
+1000cee4c6a70407e4c6a70507e4c6a70607e4c6a70707
+```
+
+**Generation of stealth box registers [[5]](#5):**
+
+```typescript
+const g = new EC("secp256k1").g; // group element generator
+const u = receiverPublicAddress;
+const r = BigInt(rand(32));
+const y = BigInt(rand(32));
+const gr = g.mul(r); // gr = g^r = R4
+const gy = g.mul(y); // gy = g^y = R5
+const ur = u.mul(r); // ur = u^r = R6
+const uy = u.mul(y); // uy = u^y = R7
+```
+
+**Box register declarations:**
+
+- Register `R4`
+  - Type: `SConstant[SGroupElement]`
+  - Value: gr = g^r
+- Register `R5`
+  - Type: `SConstant[SGroupElement]`
+  - Value: gy = g^y
+- Register `R6`
+  - Type: `SConstant[SGroupElement]`
+  - Value: ur = u^r
+- Register `R7`
+  - Type: `SConstant[SGroupElement]`
+  - Value: ur = u^r
+
+> 📝 As discussed in the `ergoforum` discussion [[1]](#1) this register declaration is larger in size than that originally proposed by `kushti` but possesses the useful property that it could look like a legitimate use-case.
+
+## `Receiver`s public address and wallets
```

Comment:
Maybe better ways of doing this but the main intent is:

1. Enable stealth boxes to be retrieved by `Receiver` after a fresh wallet restore
2. Introduce very minimal friction to `Sender` when making a transaction (ideally no change in the user journey when making a payment)

Reply:
I can't see a way of including the pubkey in the ErgoTree body while preserving privacy (with my limited crypto experience) even with a different derivation path

For example:
1. Alice posts public key for receiving stealth payment
2. Bob makes payment to Alice including her pubkey in the ErgoTree so it can be easily looked up
3. Anyone can see that Bobs payment was to Alice because of the presence of the Alices pubkey she posted in the ErgoTree

Reply:
Exactly, so might be worth removing this from the EIP scope and describing how to test if Stealth Boxes are spendable.